src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/Completion/Delegation/DelegatedCompletionHelper.cs

```diff
@@ -207,6 +207,15 @@ public static bool ShouldIncludeSnippets(RazorCodeDocument razorCodeDocument, in
         var tree = razorCodeDocument.GetSyntaxTree();
 
         var token = tree.Root.FindToken(absoluteIndex, includeWhitespace: false);
+        if (token.Kind == SyntaxKind.EndOfFile &&
+            absoluteIndex > 0)
+        {
+            // If we're at the end of the file, we check if the previous token is an open angle (ie, we're in <$$[EOF])
+            // because even though the tree won't match our expectations for an incomplete start tag, thats what it is.
```

Comment:
Would it make sense to modify RazorSyntaxFacts.IsAnyStarTag instead?

Reply:
Even if we did that, it wouldn't solve the issue for this because the final check in this method checks the span of the start tag, and that span doesn't include the end of file. Unless the compiler changes the shape of the Razor syntax tree I think this is always going to need some kind of special-casing.